extensions/sql-database-projects/src/common/httpClient.ts

```diff
@@ -48,4 +54,54 @@ export class HttpClient {
 		}
 		return response.data;
 	}
+
+	/**
+	 * Gets a file/fileContents at the given URL.
+	 * @param downloadUrl The URL to download the file from
+	 * @param targetPath The path to download the file to
+	 * @param outputChannel The output channel to output status messages to
+	 * @returns Full path to the downloaded file or the contents of the file at the given downloadUrl
+	 */
+	public download(downloadUrl: string, targetPath: string, outputChannel?: vscode.OutputChannel): Promise<void> {
```

Comment:
There's gotta be a module that does downloading with progress reporting that we can use instead...

Reply:
...and yet I can't find one.  How does this not exist?

Reply:
yeah it's surprising. Both the machine learning and azcli extensions in ADS have also implemented this function. Probably would be good to make a module ourselves that does this since it's being copied into so many extensions now...